src/ray/raylet/worker_pool.cc

```diff
@@ -1180,22 +1180,23 @@ void WorkerPool::WarnAboutSize() {
     for (const auto &starting_process : state.starting_worker_processes) {
       num_workers_started_or_registered += starting_process.second.num_starting_workers;
     }
+    // Don't count IO workers towards the warning message threshold.
+    num_workers_started_or_registered -= RayConfig::instance().max_io_workers() * 2;
```

Comment:
Why don't we just filter the io worker from the code above than doing this? Isn't it possible this will generate the false positives? (e.g., there's 0 io worker started, so, the warning is not printed for a long time).

Reply:
For simplicity. This is only erring on the side of printing the warning less (and by a small amount), so it seems worth it compared to the complexity of checking which starting workers are IO worker type, etc.

Reply:
Hmm I see. I thought there's already an available field to access these data (so it is 1~2 lines changes), but if not, I agree this solution is good enough with simplicity